src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -0,0 +1,191 @@
+// Libraries
+import React, {FC, useCallback, useState} from 'react'
+import {fromFlux} from '@influxdata/giraffe'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
+
+// Utils
+import {postWrite} from 'src/client'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+import {getByID} from 'src/resources/selectors'
+
+// Types
+import {AppState, Bucket, ResourceType, WritePrecision} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface CsvUploaderContextType {
+  handleDismiss: () => void
+  handleDrop: (csv: string) => void
+  hasFile: boolean
+  uploadFinished: boolean
+  value: number
+}
+
+export const DEFAULT_CONTEXT: CsvUploaderContextType = {
+  handleDismiss: () => {},
+  handleDrop: (_: string) => {},
+  hasFile: false,
+  uploadFinished: false,
+  value: 0,
+}
+
+export const CsvUploaderContext = React.createContext<CsvUploaderContextType>(
+  DEFAULT_CONTEXT
+)
+
+const MAX_CHUNK_SIZE = 1750
+
+export const CsvUploaderProvider: FC<Props> = React.memo(({children}) => {
+  const [value, setValue] = useState(0)
+  const {bucketID, orgID} = useParams()
+  const [hasFile, setHasFile] = useState(false)
+  const [uploadFinished, setUploadFinished] = useState(false)
+
+  const bucket =
+    useSelector((state: AppState) =>
+      getByID<Bucket>(state, ResourceType.Buckets, bucketID)
+    )?.name ?? ''
+  const org = useSelector(getOrg).name
+
+  const history = useHistory()
+
+  const handleDismiss = useCallback(() => {
```

Comment:
it'd be better not to use any route based eventing and instead use a popup controller